datarepo-clienttests/src/main/java/common/utils/KubernetesClientUtils.java

```diff
@@ -337,50 +339,66 @@ public static void changeReplicaSetSizeAndWait(int podCount) throws Exception {
     logger.debug("Pod Count: {}; Message: After scaling pod count", apiPodCount);
-    printApiPods(apiDeployment);
+    logPodsWithLabel(apiDeployment);
   }
 
-  private static long getApiPodCount(V1Deployment deployment) throws ApiException {
+  /**
+   * Return a stream of the pods that match the deployment component label.
+   *
+   * @param deployment the deployment the pods belong to
+   * @return stream of matching pods
+   */
+  private static Stream<V1Pod> listPodsWithLabel(V1Deployment deployment) throws ApiException {
     String deploymentComponentLabel = deployment.getMetadata().getLabels().get(componentLabel);
     // loop through the pods in the namespace
     // find the ones that match the deployment component label (e.g. find all the API pods)
-    long apiPodCount =
-        listPods().stream()
-            .filter(
-                pod ->
-                    deploymentComponentLabel.equals(
-                        pod.getMetadata().getLabels().get(componentLabel)))
-            .count();
-    return apiPodCount;
+    return listPods().stream()
+        .filter(
+            pod ->
+                deploymentComponentLabel.equals(pod.getMetadata().getLabels().get(componentLabel)));
   }
 
-  private static long getApiReadyPods(V1Deployment deployment) throws ApiException {
-    String deploymentComponentLabel = deployment.getMetadata().getLabels().get(componentLabel);
-    long apiPodCount =
-        listPods().stream()
-            .filter(
-                pod ->
-                    deploymentComponentLabel.equals(
-                            pod.getMetadata().getLabels().get(componentLabel))
-                        && pod.getStatus().getContainerStatuses().get(0).getReady())
-            .count();
-    return apiPodCount;
+  /**
+   * Count the number of pods that match the deployment component label.
+   *
+   * @param deployment the deployment the pods belong to
+   * @return the number of matching pods
+   */
+  private static long countPodsWithLabel(V1Deployment deployment) throws ApiException {
+    return listPodsWithLabel(deployment).count();
   }
 
-  public static void printApiPods(V1Deployment deployment) throws ApiException {
-    String deploymentComponentLabel = deployment.getMetadata().getLabels().get(componentLabel);
-    listPods().stream()
+  /**
+   * Count the number of pods that match the deployment component label and have the READY state.
+   *
+   * @param deployment the deployment the pods belong to
+   * @return the number of matching READY pods
+   */
+  private static long countReadyPodsWithLabel(V1Deployment deployment) throws ApiException {
+    return listPodsWithLabel(deployment)
         .filter(
             pod ->
-                deploymentComponentLabel.equals(pod.getMetadata().getLabels().get(componentLabel)))
-        .forEach(p -> logger.debug("Pod: {}", p.getMetadata().getName()));
+                pod.getStatus().getContainerStatuses()
```

Comment:
FWIW, I tend to like optionals for this kind of operation.  I find it's a bit more concise:
```Optional.ofNullable(pod.getStatus()).map(s -> s.getContainerStatuses().get(0).getReady().orElse(false)``